addons/hr_holidays/views/hr_leave_views.xml

```diff
@@ -421,14 +444,36 @@
-                    <field name="multi_employee" invisible="1" force_save="1"/>
-                    <field name="employee_id" groups="hr_holidays.group_hr_holidays_user" invisible="holiday_type != 'employee' or state != 'validate' or not employee_id" readonly="state in ['cancel', 'refuse', 'validate', 'validate1']" widget="many2one_avatar_user"/>
-                    <field name="employee_ids" groups="hr_holidays.group_hr_holidays_user" invisible="holiday_type != 'employee' or (state == 'validate' and employee_id)" readonly="state in ['cancel', 'refuse', 'validate', 'validate1']" required="holiday_type == 'employee' and state in ('draft', 'cancel', 'refuse')" widget="many2many_tags_avatar"/>
+                    groups="hr_holidays.group_hr_holidays_user"
+                    readonly="state not in ['confirm', 'draft']"/>
+                <field name="mode_company_id" string="Company"
+                    groups="hr_holidays.group_hr_holidays_user"
+                    invisible="holiday_type != 'company'"
+                    readonly="state not in ['confirm', 'draft']"
+                    required="holiday_type == 'company'"/>
+                <field name="category_id" groups="hr_holidays.group_hr_holidays_user"
+                    invisible="holiday_type != 'category'"
+                    readonly="state not in ['confirm', 'draft']"
+                    required="holiday_type == 'category'"/>
+                <field name="department_id" groups="hr_holidays.group_hr_holidays_user"
+                    invisible="holiday_type != 'department'"
+                    readonly="state in ['cancel', 'refuse', 'validate', 'validate1']"
+                    required="holiday_type == 'department'"/>
+                <field name="multi_employee" invisible="1" force_save="1"/>
+                <field name="allowed_employee_ids" invisible="1"/>
+                <field name="employee_id" groups="hr_holidays.group_hr_holidays_user"
+                    domain="[('id', 'in', allowed_employee_ids)]"
+                    invisible="holiday_type != 'employee' or state != 'validate' or not employee_id"
+                    readonly="state in ['cancel', 'refuse', 'validate', 'validate1']"
+                    widget="many2one_avatar_user"/>
+                <field name="employee_ids" groups="hr_holidays.group_hr_holidays_user"
+                    domain="[('id', 'in', allowed_employee_ids)]"
+                    invisible="holiday_type != 'employee' or (state == 'validate' and employee_id)"
+                    readonly="state in ['cancel', 'refuse', 'validate', 'validate1']"
+                    required="holiday_type == 'employee' and state in ('draft', 'cancel', 'refuse')"
+                    widget="many2many_tags_avatar"/>
```

Comment:
In the future I want to see changes like this split up it two commits :innocent:  one functional and one stylistic, because I have to trace with the finger to see what is that changes, while git would have highlighted what is added